internal/dataplane/parser/errors.go

```diff
@@ -0,0 +1,75 @@
+package parser
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// TranslationError represents an error occurring during translating Kubernetes objects into Kong ones.
+// It can be associated with one or more Kubernetes objects.
+type TranslationError struct {
+	causingObjects []client.Object
+	reason         string
+}
```

Comment:
any reason not to implement an error interface?

Reply:
There was no reason to implement that, and probably won't. Should we rename it to not confuse with the `Error` suffix, or are you ok with leaving it as it is?

Reply:
For me, rename would be preferable, but if others are OK with the name I'm fine too.

Reply:
Yeah, I think that makes sense. Let's save others from confusion in the future: 
29ea2ba